pkgs/development/python-modules/kikit/default.nix

```diff
@@ -0,0 +1,87 @@
+{ bc
+, zip
+, lib
+, fetchFromGitHub
+, bats
+, buildPythonPackage
+, pythonOlder
+, callPackage
+, kicad
+, numpy
+, click
+, markdown2
+, commentjson
+, wxPython_4_2
+, pcbnew-transition
+, pybars3
+, pytest
+, versioneer
+}:
+let
+  properCaseName = "KiKit";
+  solidpython = callPackage ./solidpython { };
+
+  # https://github.com/yaqwsx/KiKit/issues/574
+  # copy-pasted from nixpkgs#8d8e62e74f511160a599471549a98bc9e4f4818d
+  shapely = callPackage ./shapely { };
+in
+buildPythonPackage rec {
+  pname = lib.toLower properCaseName;
+  version = "1.3.0";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchFromGitHub {
+    owner = "yaqwsx";
+    repo = properCaseName;
+    rev = "v${version}";
+    hash = "sha256-kDTPk/R3eZtm4DjoUV4tSQzjGQ9k8MKQedX4oUXYzeo=";
+  };
+
+  propagatedBuildInputs =
+    [
+      kicad
+      numpy
+      click
+      markdown2
+      commentjson
+      # https://github.com/yaqwsx/KiKit/issues/575
+      wxPython_4_2
+      pcbnew-transition
+      pybars3
+      shapely
+      # https://github.com/yaqwsx/KiKit/issues/576
+      solidpython
+    ];
+
+  nativeBuildInputs = [
+    versioneer
+    bc
+    zip
+  ];
+
+  nativeCheckInputs = [
+    pytest
+    bats
+  ];
+
+  pythonImportsCheck = [
+    "kikit"
+  ];
+
+  checkPhase = ''
+    runHook preCheck
+    export PATH=$PATH:$out/bin
+    make test
+    runHook postCheck
```

Comment:
We should run the pytest tests with pytestCheckHook and change the directory in preCheck. The bats tests should probably be run in preCheck or postCheck then.

Reply:
Marking resolved as this is done.